src/generate-rsa-keypair/main.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2018-2022 MobileCoin Inc.
+
+//! Utility to generate a 4096-bit passphrase-less RSA keypair, meant to be used for private<->client end to end encryption.
+
+use boring::rsa::Rsa;
+use std::{fs, path::Path};
+
+const PRIVATE_KEY_FILENAME: &str = "mirror-private.pem";
+const PUBLIC_KEY_FILENAME: &str = "mirror-client.pem";
+
+fn main() {
+    if Path::new(PRIVATE_KEY_FILENAME).exists() {
```

Comment:
If the private exists, but the public does not, is it possible to generate the public from the private? I'm not sure if that would ever be useful though.

Reply:
Yes, it is possible. I'll change it to try and read the private key if it exists.